pkg/setting/setting_secure_socks_proxy.go

```diff
@@ -40,5 +43,45 @@ func readSecureSocksDSProxySettings(iniFile *ini.File) (SecureSocksDSProxySettin
 		return s, errors.New("proxy address required")
 	}
 
+	err := SetProxyEnvVariables(s)
+	if err != nil {
+		return s, err
+	}
+
 	return s, nil
 }
+
+// SetProxyEnvVariables sets the needed environment variables for grafana core datasoruces
+// to use the grafana-plugin-sdk code to set up the proxy
+func SetProxyEnvVariables(settings SecureSocksDSProxySettings) error {
```

Comment:
This feels a bit fragile and non-optimal. Not sure what the alternative is though :cry: We've had similar problems earlier the SDK using lots of global state, e.g. the [logger](https://github.com/grafana/grafana-plugin-sdk-go/blob/990631eb75f2f55898fa442c279c6073372015c8/backend/log/log.go#L94) needing to have different implementations in Grafana vs plugin. 

The optimal thing would be if we could override the transport in https://github.com/grafana/grafana/blob/main/pkg/infra/httpclient/httpclientprovider/http_client_provider.go#L21, but tricky due to how `httpclient.GetTransport` implements secure proxying ([code](https://github.com/grafana/grafana-plugin-sdk-go/blob/990631eb75f2f55898fa442c279c6073372015c8/backend/httpclient/http_client.go#L89-L92)).  

Maybe introduce `httpclient.Options.SecureSocksDisabled` which is default false and only register secure socks when false. In Grafana we can provide the option to be true. But there's already [ProxyOptions](https://github.com/grafana/grafana-plugin-sdk-go/blob/990631eb75f2f55898fa442c279c6073372015c8/backend/httpclient/options.go#L36) in there so it might be weird :shrug: 

Another alternative expose the secure socks config as a global variable in the SDK which by default reads from the env, but Grafana can override. Would be similar idea as the global logger above. Maybe this is not something we want to allow from a security perspective?

Reply:
I agree - thanks for thinking through various alternatives that could be used! I'll look into exposing the secure socks config as a global variable